auction-app-backend/src/main/java/com/atlantbh/auctionappbackend/domain/User.java

```diff
@@ -0,0 +1,157 @@
+package com.atlantbh.auctionappbackend.domain;
+
+import javax.persistence.*;
+import java.time.LocalDate;
+
+@Entity(name = "User")
+@Table(
+        name = "app_user",
+        uniqueConstraints = {
+                @UniqueConstraint(name = "user_email_unique", columnNames = "email")
+        }
+)
+public class User {
+    @Id
+    @SequenceGenerator(
+            name = "app_user_sequence",
+            sequenceName = "app_user_sequence",
+            allocationSize = 1
+    )
+    @GeneratedValue(
+            strategy = GenerationType.SEQUENCE,
+            generator = "app_user_sequence"
+    )
+    @Column(updatable = false)
+    private Long id;
+
+    @Column(nullable = false)
+    private String firstName;
+
+    @Column(nullable = false)
+    private String lastName;
+
+    @Column(nullable = false)
+    private String email;
+
+    @Column(nullable = false)
+    private String password;
+
+    @Column(nullable = false)
+    private String role;
+
+    private Long streetId;
+    private LocalDate dateOfBirth;
+    private String phoneNumber;
+    private String photoUrl;
+
+    public User() {
+    }
```

Comment:
Is empty constructor necessary? If not, remove it.

Reply:
Probably not. There are no references to it. Since it is an Entity which will be mapped to a table in the database, is there any reason to keep it for those purposes?

Reply:
If the empty constructor is removed, there is an error when trying to create user via the register user api.
`com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of 'com.atlantbh.auctionappbackend.domain.User' (no Creators, like default constructor, exist): cannot deserialize from Object value (no delegate- or property-based Creator)`